packages/deployer/internal/ast/router-ast-validator.js

```diff
@@ -30,14 +30,43 @@ async function findMissingSelectorsInAST(contracts) {
   return errors;
 }
 
-async function findUnreachableSelectorsInAST() {
-  logger.info('  Unreachable selectors not checked yet in compiled router');
-  return [];
+async function findUnreachableSelectorsInAST(contracts, modules) {
+  const routerSelectors = getCaseSelectors('Router', contracts['Router']);
+  const moduleAddresses = [];
+  for (const [moduleName, moduleData] of Object.entries(modules)) {
+    moduleAddresses[toPrivateConstantCase(moduleName)] = moduleData.deployedAddress;
+  }
+
+  const errors = [];
+  routerSelectors.forEach((s) => {
+    if (!moduleAddresses[s.value.name] || moduleAddresses[s.value.name] !== s.value.value.value) {
```

Comment:
Checks that the address of the contract pointed is the one of the contract.

Reply:
Could this also check that the hardcoded module name is correct?

Right now of you change for example:
`case 0xa40674b7 { result := _SOME_MODULE } // SomeModule.getSomeValue()`
to
`case 0xa40674b7 { result := _OWNER_MODULE } // SomeModule.getSomeValue()`

It does not fail.

Reply:
good catch. It doesn't seem that trivial, I might create another issue for that.

Reply:
done